tests/kas_messaging.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { test, expect } from '@playwright/test';
+import login from '@lib/auth';
+import { config } from '@lib/config';
+import {
+  navigateToKafkaList,
+  deleteKafkaInstance,
+  createKafkaInstance,
+  getBootstrapUrl,
+  navigateToAccess,
+  navigateToConsumerGroups,
+  grantProducerAccess,
+  grantConsumerAccess
+} from '@lib/kafka';
+import { navigateToKafkaTopicsList, createKafkaTopic, navigeToMessages } from '@lib/topic';
+import { KafkaConsumer, KafkaProducer } from '../lib/clients';
+import { strict as assert } from 'assert';
+import { createServiceAccount, deleteServiceAccount, navigateToSAList } from '@lib/sa';
+
+const testInstanceName = 'test-instance-messaging';
+const testTopicName = `test-topic-name`;
+const testServiceAccountName = 'test-service-account';
+const testMessageKey = 'key';
+const consumerGroupId = 'test-consumer-group';
+const expectedMessageCount = 100;
+let credentials;
+let bootstrap: string;
+
+test.beforeEach(async ({ page }) => {
+  await login(page);
+
+  await navigateToKafkaList(page);
+
+  await expect(page.getByRole('button', { name: 'Create Kafka instance' })).toBeVisible();
+
+  if ((await page.getByText(testInstanceName).count()) > 0 && (await page.locator('tr').count()) === 2) {
+    // Test instance present, nothing to do!
+  } else {
+    await page.waitForSelector('[role=progressbar]', {
+      state: 'detached',
+      timeout: config.kafkaInstanceCreationTimeout
+    });
+
+    for (const el of await page.locator(`tr >> a`).elementHandles()) {
+      const name = await el.textContent();
+
+      if (name !== testInstanceName) {
+        await deleteKafkaInstance(page, name);
+      }
+    }
+
+    if ((await page.getByText(testInstanceName).count()) === 0) {
+      await createKafkaInstance(page, testInstanceName);
+    }
+
+    await navigateToKafkaTopicsList(page, testInstanceName);
```

Comment:
I see that this can work, but I would feel more comfortable if each step of the logic from here to the end is performed "always" (e.g. outside of the `else` starting at line 37) using appropriate checks to identify the existence of each and every element we are creating (there are already some in place).

This will make it easier to handle while debugging, wdyt?

Reply:
Yes, that's a very good catch!